### ISSUE TITLE
[Docs] parameter data_batch is not used

### ISSUE THREAD

**Issue by YQisme** [open-mmlab/mmengine#950]
### 📚 The doc issue

```python
from mmengine.evaluator import BaseMetric

class Accuracy(BaseMetric):
    def process(self, data_batch, data_samples):
        score, gt = data_samples
        # save the middle result of a batch to `self.results`
        self.results.append({
            'batch_size': len(gt),
            'correct': (score.argmax(dim=1) == gt).sum().cpu(),
        })

    def compute_metrics(self, results):
        total_correct = sum(item['correct'] for item in results)
        total_size = sum(item['batch_size'] for item in results)
        # return the dict containing the eval results
        # the key is the name of the metric name
        return dict(accuracy=100 * total_correct / total_size)
```
parameter `data_batch` is not used
https://mmengine.readthedocs.io/en/latest/get_started/15_minutes.html#build-a-evaluation-metrics

**Comment by C1rN09** [comment]
**TL;DR: `data_batch` is required in `BaseMetric` api, but whether you use it depends on your model implementation.**

This is relevant to the dataflow in MMEngine. To be simple, `data_batch` is the raw data obtained from `dataloader`, while `data_samples` is the prediction of your model. In this example the model prediction contains gt, so you can leave `data_batch` unused.

For more details, you can refer to the [Runner Tutorial](https://mmengine.readthedocs.io/en/latest/tutorials/runner.html#basic-dataflow) or [ValLoop.run_iter](https://github.com/open-mmlab/mmengine/blob/8d4885cb4325be88eb2ad75510636d281bc73499/mmengine/runner/loops.py#L368)
